src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TreeHasher.java

```diff
@@ -91,6 +92,12 @@ public void visitLiteral(JCLiteral tree) {
         super.visitLiteral(tree);
     }
 
+    @Override
+    public void visitClassDef(JCClassDecl tree) {
+        hash(tree.sym);
+        super.visitClassDef(tree);
```

Comment:
I would remove the recursive call here - the hash of a class is just that of its symbol - no need to scan its members recursively.